src/highspy/highs.py

```diff
@@ -106,7 +106,9 @@ def update(self):
         current_batch_size = len(self._batch.obj)
         print("\nDEBUG update(self): On entry - current_batch_size = ", current_batch_size)
         if current_batch_size > 0:
-            names = [self._vars[i].name for i in range(current_batch_size)]
+            # names = [self._vars[i].name for i in range(current_batch_size)]
+            names = [self._batch.name[i] for i in range(current_batch_size)]
```

Comment:
Yes, this fixes it. It's actually nothing to do with adding update() to addVariables, since this would be wrong once update() has been called for the first non-empty batch.